File: engine/src/com/magma/engine/debug/modules/InfoModule.java
package com.magma.engine.debug.modules;

import javax.swing.JLabel;
import com.badlogic.gdx.Gdx;

public class InfoModule extends Module {

    private final JLabel fps, mem;

    public InfoModule() {
        super("Info");

        fps = new JLabel();
        mem = new JLabel();

        add(fps);
        add(mem);
    }

    @Override
	public void update() {
	    fps.setText(format("FPS",Gdx.graphics.getFramesPerSecond()));
        mem.setText(format("Memory",getMemory()));
	}

    public String getMemory() {
        final int mb = 1024 * 1024;
        long mem = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / mb;
        long totalMem = Runtime.getRuntime().totalMemory() / mb;
        long maxMem = Runtime.getRuntime().maxMemory() / mb;
        return Long.toString(mem) + " | reserved " + Long.toString(totalMem) + " | cap " + Long.toString(maxMem)
                + " mb";
    }
}

File: engine/src/com/magma/engine/debug/modules/InfoModule.kt
package com.magma.engine.debug.modules

import com.badlogic.gdx.Gdx
import javax.swing.JLabel

class InfoModule : Module("Info") {
    private val fps: JLabel = JLabel()
    private val mem: JLabel = JLabel()

    init {
        add(fps)
        add(mem)
    }

    override fun update() {
        fps.text = format("FPS", Gdx.graphics.framesPerSecond)
        mem.text = format("Memory", memory)
    }

    private val memory: String
        get() {
            val mb = 1024 * 1024
            val mem = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / mb
            val totalMem = Runtime.getRuntime().totalMemory() / mb
            val maxMem = Runtime.getRuntime().maxMemory() / mb
            return (mem.toString() + " | reserved " + totalMem.toString() + " | cap " + maxMem.toString()
                    + " mb")
        }
}